gnovm/stdlibs/internal/stdgen/stdgen.go

```diff
@@ -0,0 +1,390 @@
+			continue
+		}
+		first := fn.Type.Params.List[0]
+		if len(first.Names) > 1 {
+			continue
+		}
+		ind, ok := first.Type.(*ast.StarExpr)
+		if !ok {
+			continue
+		}
+		res, ok := ind.X.(*ast.SelectorExpr)
+		if !ok {
+			continue
+		}
+		if id, ok := res.X.(*ast.Ident); ok && id.Name == iname && res.Sel.Name == "Machine" {
+			id.Name = "#gnomachine"
+		}
+	}
+	return fns
+}
+
+func gnolangImportName(f *ast.File) string {
+	for _, i := range f.Imports {
+		ipath, err := strconv.Unquote(i.Path.Value)
+		if err != nil {
+			continue
+		}
+		if ipath == "github.com/gnolang/gno/gnovm/pkg/gnolang" {
+			if i.Name == nil {
+				return "gnolang"
+			}
+			return i.Name.Name
+		}
+	}
+	return ""
+}
+
+type mapping struct {
+	GnoImportPath  string
+	GnoMethod      string
+	GnoParamTypes  []string
+	GnoResultTypes []string
+	GoImportPath   string
+	GoFunc         string
+	MachineParam   bool
+}
+
+func linkFunctions(pkgs []*pkgData) []mapping {
+	var mappings []mapping
+	for _, pkg := range pkgs {
+		for _, gb := range pkg.gnoBodyless {
+			nameWant := gb.Name.Name
+			if !gb.Name.IsExported() {
+				nameWant = "X_" + nameWant
+			}
+			fn := findFuncByName(pkg.goExported, nameWant)
+			if fn == nil {
+				logWarning("package %q: no matching go function declaration (%q) exists for function %q",
+					pkg.importPath, nameWant, gb.Name.Name)
+				continue
+			}
+			mp := mapping{
+				GnoImportPath: pkg.importPath,
+				GnoMethod:     gb.Name.Name,
+				GoImportPath:  "github.com/gnolang/gno/gnovm/stdlibs/" + pkg.importPath,
+				GoFunc:        fn.Name.Name,
+			}
+			if !mp.loadSignaturesMatch(gb, fn) {
+				logWarning("package %q: signature of gno function %s doesn't match signature of go function %s",
+					pkg.importPath, gb.Name.Name, fn.Name.Name)
+				continue
+			}
+			mappings = append(mappings, mp)
+		}
+	}
+	return mappings
+}
+
+func findFuncByName(fns []*ast.FuncDecl, name string) *ast.FuncDecl {
+	for _, fn := range fns {
+		if fn.Name.Name == name {
+			return fn
+		}
+	}
+	return nil
+}
+
+func (m *mapping) loadSignaturesMatch(gnof, gof *ast.FuncDecl) bool {
+	if gnof.Type.TypeParams != nil || gof.Type.TypeParams != nil {
+		panic("type parameters not supported")
+	}
+	// Ideally, signatures match when they accept the same types,
+	// or aliases.
+	gnop, gnor := fieldListToTypes(gnof.Type.Params), fieldListToTypes(gnof.Type.Results)
+	// store gno params and results in mapping
+	m.GnoParamTypes = gnop
+	m.GnoResultTypes = gnor
+	gop, gor := fieldListToTypes(gof.Type.Params), fieldListToTypes(gof.Type.Results)
+	if len(gop) > 0 && gop[0] == "*gno.Machine" {
+		m.MachineParam = true
+		gop = gop[1:]
+	}
+	return reflect.DeepEqual(gnop, gop) && reflect.DeepEqual(gnor, gor)
+}
+
+var builtinTypes = [...]string{
+	"bool",
+	"string",
+	"int",
+	"int8",
+	"int16",
+	"rune",
+	"int32",
+	"int64",
+	"uint",
+	"byte",
+	"uint8",
+	"uint16",
+	"uint32",
+	"uint64",
+	"bigint",
+	"float32",
+	"float64",
+	"error",
+	"any",
```

Comment:
I suppose that this list is just for the PoC. That info is redundant and it depends on the implemented types, so would be great to only have one source of truth for this.